pkgs/tools/text/shfmt/default.nix

```diff
@@ -2,16 +2,16 @@
 
 buildGoModule rec {
   pname = "shfmt";
-  version = "3.0.2";
+  version = "3.1.0";
 
   src = fetchFromGitHub {
     owner = "mvdan";
     repo = "sh";
     rev = "v${version}";
-    sha256 = "1q0gazh87y7sl5sl5m046a83d64aas9xnbg2d1d1h2vwcqdaccp2";
+    sha256 = "1k0bq4b4rv6wrh24jvcnpg1mfqrzqhl90zg4zibxzv3zkhjdskzh";
   };
 
-  modSha256 = "1ll2cxhgf8hh19wzdykgc81c4yfcp8bzmfaif08nvvb63rhjdb5y";
+  modSha256 = "080k8d5rp8kyg0x7vjxm758b9ya9z336yd4rcqws7yhqawxiv55z";
   subPackages = ["cmd/shfmt"];
 
```

Comment:
would you mind adding the following?
```
  buildFlagsArray = [ "-ldflags=-s -w -X main.version=${version}" ];
```

Reply:
Upstream seems to have broken this, it wasn't needed previously and even 3.1.0 pre-built binaries don't set the version correctly.

Reply:
is not broken per se, upstream set [`version` to devel](https://github.com/mvdan/sh/commit/bb3fed3b98a2660cc3e7371a37ce79634d609f08), and allows to write `main.version` at link time, see its dockerfile
https://github.com/mvdan/sh/blob/v3.1.0/cmd/shfmt/Dockerfile#L5

Reply:
> https://github.com/mvdan/sh/commit/bb3fed3b98a2660cc3e7371a37ce79634d609f08
> cmd/shfmt: set version to "devel" in master 
> Otherwise building master might feel like it's a release version.
> Releases are tagged, and those tags have the right version.

> https://github.com/mvdan/sh/commit/8e403df550b5c4f162f1fd5b5c803d49a0ee8b63 
> cmd/shfmt: start using module info for -version 
> This way, 'go get' installs will automatically have proper version
information. Do it in a way that doesn't break -ldflags=-X, since we use
that in the dockerfile.

Reply:
I've added `ldfags`, we can remove it in a future release once it is fixed.

Reply:
Thanks for humor me. I must recognize that I barely know Golang, and you are probably right about this being a upstream issue